hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpCommand.java

```diff
@@ -56,14 +55,29 @@
     protected final String uri;
     protected ByteBuffer response;
     protected boolean nextLine;
+    protected final RestCallExecutionListener listener;
 
+    protected HttpCommand(TextCommandConstants.TextCommandType type, String uri, ServerConnection connection) {
+        this(type, uri, connection.getConnectionManager().getServer().getContext()
+                .getTextCommandService().createRestCallExecutionListener());
+    }
 
-    public HttpCommand(TextCommandConstants.TextCommandType type, String uri) {
+    public HttpCommand(TextCommandConstants.TextCommandType type, String uri, RestCallExecutionListener listener) {
```

Comment:
From the usages of this constructors, I could not understand why a second constructor is needed. It looks like only the top one is called from outside. Can't we merge them to one constructor ?